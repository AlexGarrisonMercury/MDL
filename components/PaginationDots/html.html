<div class="pagination-dots">
  <div class="pagination-dots__dot"></div>
  <div class="pagination-dots__dot--filled"></div>
</div>
