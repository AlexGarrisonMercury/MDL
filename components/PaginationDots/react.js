import React, { memo } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames/bind'
import style from './style.module.scss'

const cx = classNames.bind(style)

export const Dot = ({ filled }) => (
  <div className={cx(['pagination-dots__dot', filled && 'pagination-dots__dot--filled'])} />
)

export const PaginationDots = ({ activePage, totalPages }) => (
  <div className="pagination-dots">
    {[...Array(totalPages).keys()].map((dot, i) => (
      <Dot key={dot} filled={i === activePage} />
    ))}
  </div>
)

export default memo(PaginationDots)

PaginationDots.propTypes = {
  activePage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired,
}

Dot.propTypes = {
  filled: PropTypes.bool.isRequired,
}
