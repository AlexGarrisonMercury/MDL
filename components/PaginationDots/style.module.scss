@import "styles/index";

.pagination-dots {
  display: inline-block;
  font-size: 0;

  &__dot {
    display: inline-block;
    height: 8px;
    width: 8px;
    margin: 4px;
    border-radius: 50%;
    background-color: $color-silver;

    &--filled {
      background-color: $color-limerick;
    }
  }
}
