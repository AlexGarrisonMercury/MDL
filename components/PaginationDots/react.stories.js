import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs, number } from '@storybook/addon-knobs'
import PaginationDots from './react'
import Readme from './react.README.md'

storiesOf('PaginationDots', module)
  .addParameters({ readme: { sidebar: Readme } })
  .addDecorator(withKnobs)
  .add('default', () => {
    const activePage = number('Active Page', 1)
    const totalPages = number('Total Pages', 5)

    return <PaginationDots activePage={activePage} totalPages={totalPages} />
  })
