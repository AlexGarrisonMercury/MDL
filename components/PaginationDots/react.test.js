import React from 'react'
import { shallow } from 'enzyme'
import PaginationDots, { Dot } from './react'

describe('PaginationDots', () => {
  let props
  let wrapper

  beforeEach(() => {
    props = {
      activePage: 2,
      totalPages: 5,
    }

    wrapper = shallow(<PaginationDots {...props} />)
  })

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should only have one active page', () => {
    expect(wrapper.findWhere(n => n.props().filled).length).toBe(1)
  })

  it('should have a filled prop set to true on the correct dot', () => {
    expect(
      wrapper
        .find('Dot')
        .at(2)
        .props().filled,
    ).toBe(true)
  })
})

describe('Dot', () => {
  it('should have filled class applied', () => {
    const wrapper = shallow(<Dot filled />)
    expect(wrapper.hasClass('pagination-dots__dot--filled')).toBe(true)
  })
})
