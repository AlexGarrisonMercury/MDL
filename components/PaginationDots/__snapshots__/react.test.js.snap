// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationDots should render without crashing 1`] = `
<div
  className="pagination-dots"
>
  <Dot
    filled={false}
    key="0"
  />
  <Dot
    filled={false}
    key="1"
  />
  <Dot
    filled={true}
    key="2"
  />
  <Dot
    filled={false}
    key="3"
  />
  <Dot
    filled={false}
    key="4"
  />
</div>
`;
